docs/proposals/zeroconf.md

```diff
@@ -0,0 +1,151 @@
+Established under name 'freddie'
+```
+
+> **NOTE** You may need to install `avahi-utils` if you're running Debian or a derivative
+
+The service will be published to the default Zeroconf domain (`local`) and its fully-qualifiied domain-name (FQDN) is thus `freddie.local`.
+
+> **NOTE** For the purposes of what follows, while distinct, hosts (devices) and services may be considered equivalent.
+
+Then, it's possible to enumerate hosts and services discovered by Zeroconf using:
+
+```bash
+avahi-browse --all
++ wlp6s0 IPv4 freddie                                       _example._tcp        local
+```
+
+## Discovery Process
+
+The Akri discovery handler is written in Rust and uses [`zeroconf`](https://crates.io/crates/zeroconf). Some of what follows may be specific to these technologies.
+
+> **NOTE** There is a proposal to replace `zeroconf` with [`astro-dnssd`](https://crates.io/crates/astro-dnssd) as this provides cross-platform support. There are some limitations with `astro-dnssd` too that are blocking this switch.
+
+The Akri Agent is deployed to a Kubernetes cluster. Kubernetes clusters commonly run in-cluster DNS services (nowadays [`CoreDNS`](https://kubernetes.io/docs/tasks/administer-cluster/coredns/)). For this reason, the applicability of the Akri Zeroconf protocol is to devices not accessible within the cluster. The benefit of the Akri Zeroconf protocol is to make off-cluster Zeroconf-accessible hosts (devices) and services accessible to Kubernetes cluster resources (e.g. applications).
+
+For Zeroconf discovery to occur, the Agent's Pod must leverage several Zeroconf depdendencies and libraries. These depdendencies not only expand the size of the Akri Agent (~800MB) but they increase the Agent's surface area and increase the possibility of vulnerabilities.
+
+Discovery is a key functionality of Zeroconf and is straightforward to implement. See the [Browsing services](https://crates.io/crates/zeroconf#browsing-services) examples of the `zeroconf` crate.
+
+One wrinkle is that Akri expects discovery to run to completion. Akri periodically reruns discovery for a protocol. The `zeroconf` crate polls networks for hosts and services.
+
+The implementation used by the protocol is to poll for 5 seconds and report back whichever hosts and services were discovered during that window.
+
+## Broker interfacing
+
+Upon detection, the Akri Zeroconf discovery handler creates an Akri instance to represent each discovered service. The instance contains information about the service using environment variables in each broker pod, if a broker pod is specified in the Configuration.
+
+A more complete description of the `freddie` service could be:
+
+```YAML
+{
+    name: "freddie",
+    kind: "_example._tcp",
+    domain: "local",
+    host_name: "freddie.local",
+    address: "192.168.1.100", port: 8888,
+    txt: ...
+}
+```
+
+Each Broker Instance is configured with environment variables corresponding to the above value:
```

Comment:
Please change "Broker instance" to "Broker Pod"

Reply:
Fixed: https://github.com/deislabs/akri/pull/162/commits/d03cb24ad07f3bdca809f300754c25a33849231f